src/Workspaces/Core/Portable/Utilities/TaskExtensions.cs

```diff
@@ -60,8 +60,10 @@ public static T WaitAndGetResult<T>(this Task<T> task, CancellationToken cancell
         // thread.  In the future we are going ot be removing this and disallowing its use.
         public static T WaitAndGetResult_CanCallOnBackground<T>(this Task<T> task, CancellationToken cancellationToken)
         {
-            task.Wait(cancellationToken);
-            return task.Result;
+            // WaitAny waits for the task to complete with cancellation and without throwing;
+            // GetAwaiter().GetResult() is then used so that if the task failed, the exception is not wrapped in AggregateException
+            Task.WaitAny(new Task[] { task }, cancellationToken);
```

Comment:
How important is it that this be very efficient?  This is penalizing with an extra array allocation what I assume is the common path (success) for avoiding an exception throw on what I assume is the uncommon path (failure).

If it matters, you could of course switch to just letting the exception get thrown and then catching it.  Or at the very least you could special case when the task is already completed (no idea how common that is in the usage of this method), surrounding this call with `if (!task.IsCompleted) { ... }`.